### DISCUSSION TITLE
GAT, sparse adjacency matrix, out-of-memory

### DISCUSSION THREAD

**Discussion by RaphaelChristienECTL** [danielegrattarola/spektral#389] [Q&A]
Hello,
thanks for Spektral !
I have been happy to see that there is a GAT convolution layer proposed :-)

The documentation states that this layer expects dense inputs when working in batch mode.

I am using a sparse matrix for A (adjacency matrix, numpy coo matrix) and a single data loader (I guess then that 
this is not a batch mode then), as in the GAT example on the Spektral website.  

Unfortunately, I am getting an out-of-memory message when launching the model fit, while
using the model GeneralGNN runs fine(but is not fit for my purpose).

Any hint why this might happen, or what to check/examples to make it work ?

By the way, I guess using a disjoint loader would still allow for the GAT to use sparse input matrix ?

Cheers !

**Comment by danielegrattarola** [comment]
[Here is an example](https://github.com/danielegrattarola/spektral/blob/master/examples/graph_prediction/tud_gin.py) of training a model in disjoint mode with sparse adjacency matrix. 

GAT should work pretty well with sparse inputs, so maybe you are accidentally converting the adjacency to dense somewhere in your code?

Cheers

**Reply by RaphaelChristienECTL** [reply]
Thanks very much for the confirmation that GAT should work well with sparse inputs and the example !

I decomposed my graph (90K nodes with 400 features /200K edges) into multiple distinct non-connected components as subgraphs and train the GAT model on the subgraphs list with a disjoint loader : no more memory complaints !
